packages/duckdb-wasm-shell/src/utils/history_store.ts

```diff
@@ -21,12 +21,14 @@ export class HistoryStore {
     protected _idb: IDBDatabase | null;
     protected _nextEntryKey: number;
     protected _entryCount: number;
+    protected _isWorking: boolean;
 
     public constructor() {
         this._idbFactory = window.indexedDB;
         this._idb = null;
         this._nextEntryKey = 0;
         this._entryCount = 0;
+        this._isWorking = (this._idbFactory != null);
```

Comment:
Instead of this state variable, would it make sense to return a different instance that has a different implementation of the methods that are not supported?

Reply:
Thanks for the review!
After fixing it, I think actually the minimal set of changes we could get away with for fixing Firefox Incognito behaviour are:
```
            req.onerror = err => {
                console.log("Error while opeing indexedDB connection", err);
                // NOTE: we are using resolve(null) on purpose, since null is a vaild value,
		//	and reject (un-catched) will trigger a failure
                resolve(null);
            };
```
instead of
```
req.onerror = err => reject(err);
```
+ doing the checks on `loadMetadata`, `reset` and `push` on _idb being initialzied (instead of adding a variable).

Reply:
Basically applying this commit: https://github.com/duckdb/duckdb-wasm/compare/master...carlopi:duckdb-wasm:fixIncognito2 instead of the current PR.

Reply:
I don't think I like either as they are both swallowing errors or intransparently change behavior. 

Can we just wrap the call where we initialize the history db to either return an instance of this class or an empty mock?